scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -198,11 +198,14 @@
   [{:keys [name]} instance-id ^V1PodStatus pod-status ^V1ContainerStatus container-status]
   ; TODO map additional kubernetes failure reasons
   (let [container-terminated-reason (some-> container-status .getState .getTerminated .getReason)
-        pod-status-reason (.getReason pod-status)]
+        pod-status-reason (.getReason pod-status)
+        pod-status-message (.getMessage pod-status)]
     (cond
       (= "OutOfMemory" pod-status-reason) :reason-container-limitation-memory
       (= "Error" container-terminated-reason) :reason-command-executor-failed
       (= "OOMKilled" container-terminated-reason) :reason-container-limitation-memory
+      (or (str/includes? pod-status-message "Pod ephemeral local storage usage exceeds the total limit of containers")
+          (str/includes? pod-status-message "The node was low on resource: ephemeral-storage")) :reason-container-limitation-disk
```

Comment:
Should we make these configurable? Not sure if it's worth the trouble -- how likely are they to change? We could alternatively make the substring slightly shorter to (maybe?) reduce the likelihood of changes breaking this?

e.g.

```
ephemeral local storage usage exceeds
low on resource: ephemeral-storage
```

It would be nice to add a comment indicating which of these is for using more than your limit versus using more than your request.

Reply:
added comments and made the substrings shorter -- not sure if its worth making it configurable but will check with mcf